offchain/indexer/src/conversions.rs

```diff
@@ -14,18 +14,7 @@ use rollups_events::{
 use rollups_data::{Input, Notice, OutputEnum, Proof, Report, Voucher};
 
 pub fn convert_input(input: RollupsAdvanceStateInput) -> Input {
-    let timestamp = match NaiveDateTime::from_timestamp_millis(
-        input.metadata.timestamp as i64,
-    ) {
-        Some(timestamp) => timestamp,
-        None => {
-            tracing::warn!(
-                input.metadata.timestamp,
-                "failed to parse timestamp"
-            );
-            Default::default()
-        }
-    };
+    let timestamp = UNIX_EPOCH + Duration::from_secs(input.metadata.timestamp);
```

Comment:
Why not only assign input.metadata.timestamp?

Reply:
In the database schema, we are using the timestamp type to represent the input timestamp. In Diesel (the ORM that we use), this type needs to be [`SystemTime`](https://doc.rust-lang.org/std/time/struct.SystemTime.html#) from the standard lib or `NaiveDateTime` from chrono. This operation converts from an integer to `SystemTime`.

> and using UNIX_EPOCH + duration can be used to create a SystemTime instance to represent another fixed point in time.

https://doc.rust-lang.org/std/time/struct.SystemTime.html#associatedconstant.UNIX_EPOCH

Reply:
Here is the list of compatible Rust types for the SQL timestamp: https://docs.diesel.rs/2.0.x/diesel/sql_types/struct.Timestamp.html